src/wallet/wallet.h

```diff
@@ -1356,6 +1363,19 @@ class CWallet final : public FillableSigningProvider, private interfaces::Chain:
 
     /** Implement lookup of key origin information through wallet key metadata. */
     bool GetKeyOrigin(const CKeyID& keyid, KeyOriginInfo& info) const override;
+
+    /** Get last block processed height */
+    int GetLastBlockHeight() const EXCLUSIVE_LOCKS_REQUIRED(cs_wallet)
+    {
+        AssertLockHeld(cs_wallet);
+        return m_last_block_processed_height;
+    };
+    /** Set last block processed height, currently only use in unit test */
+    void SetLastBlockHeight(int block_height) EXCLUSIVE_LOCKS_REQUIRED(cs_wallet)
```

Comment:
3d32a691d2df4c73a39a64a5d331c29b5eb94213

Why isn't this named `SetLastBlockProcessedHeight`?

Anyway I think the variable should be `m_chain_tip_height`. Also, I think now `m_last_block_processed` should be improved, like:
```cpp
    uint256 m_chain_tip_hash GUARDED_BY(cs_wallet);
    int m_chain_tip_height GUARDED_BY(cs_wallet){-1};
```
and both should be set at once.

Reply:
Agree for the name of method and set at once, but not sure for variables. I think it would be confusing, as intent of this PR is to let the wallet have its own view of what chain tip is after processing block from validation interface. So real chain tip and the wallet one may diverge.

Reply:
Yeah these would be from the wallet POV.